src/common/transformations/src/transformations/common_optimizations/sdpa_fusion.cpp

```diff
@@ -155,8 +155,8 @@ SDPAFusion::SDPAFusion() {
         if (v_node_ps[-2] != S)
             return false;
 
-        // make sure that embedding dimension of key is the same as for query
-        if (k_node_ps[-1] != E)
+        // make sure that embedding dimension of key and value is the same as for query
+        if (k_node_ps[-1] != E || v_node_ps[-1] != E)
```

Comment:
As discussed offline, could you share when this can happen? Also, I think we need some test for this case.

Reply:
SDPA candidate node info:
- Query node {8,4,64,16}
- Key node {8,4,144,16}
- Value node {8,4,144,32}
AS value's head dim is not match with Query and Key, SDPA fusion is not possible